src/views/Editor/index.js

```diff
@@ -33,84 +43,98 @@ var sentinel = true;
+        this.enableButton = this.enableButton.bind(this)
+        this.disableButton = this.disableButton.bind(this)
+        this.getDeleteButton = this.getDeleteButton.bind(this)
+        this.eventExists = this.eventExists.bind(this)
+        this.getCancelButton = this.getCancelButton.bind(this)
+        this.getSaveButtons = this.getSaveButtons.bind(this)
+        this.getActionButtons = this.getActionButtons.bind(this)
+        this.clearForm = this.clearForm.bind(this)
+        this.getWarningMarkup = this.getWarningMarkup.bind(this)
+        this.saveAsDraft = this.saveAsDraft.bind(this)
+        this.saveAsPublished = this.saveAsPublished.bind(this)
+        this.confirmDelete = this.confirmDelete.bind(this)
+        this.deleteEvents = this.deleteEvents.bind(this)
+        this.deleteSubEvent = this.deleteSubEvent.bind(this)
+        this.confirmCancel = this.confirmCancel.bind(this)
+    }
 
     componentWillMount() {
         if(this.props.match.params.action === 'update' && this.props.match.params.eventId) {
-            this.props.dispatch(fetchEventForEditing(this.props.match.params.eventId, this.props.user))
-            this.props.dispatch(fetchSubEvents(this.props.match.params.eventId, this.props.user))
+            this.props.fetchEventForEditing(this.props.match.params.eventId, this.props.user)
+            this.props.fetchSubEvents(this.props.match.params.eventId, this.props.user)
         }
-    },
+    }
 
     componentDidMount() {
-        window.addEventListener("beforeunload", this.handler)
-    },
+        window.addEventListener('beforeunload', this.handler)
+    }
 
-    componentWillReceiveProps: function(nextProps) {
-        // Check if we are changing the editing mode on fly
-        // (happens when jumping from update event page to create event page)
-        // Clear page or fetch new eventdata accordingly
+    componentWillReceiveProps(nextProps) {
+    // Check if we are changing the editing mode on fly
+    // (happens when jumping from update event page to create event page)
+    // Clear page or fetch new eventdata accordingly
         if(nextProps.match && nextProps.match.params && this.props.match.params.action !== nextProps.match.params.action) {
             if(nextProps.match.params.action === 'update') {
-                this.props.dispatch(fetchEventForEditing(this.props.match.params.eventId), this.props.user)
+                this.props.fetchEventForEditing(this.props.match.params.eventId, this.props.user)
             } else {
-                this.props.dispatch(clearData())
+                this.props.clearData()
             }
         }
 
         this.forceUpdate()
-    },
+    }
 
     componentWillUnmount() {
-        window.removeEventListener("beforeunload", this.handler)
-        this.props.dispatch(setValidationErrors({}))
-    },
+        window.removeEventListener('beforeunload', this.handler)
+        this.props.setValidationErrors({})
+    }
 
     routerWillLeave(nextLocation) {
```

Comment:
I suspect this method has something to do with the react-router `Lifecycle` mixin that no longer exists. So this method could probably be removed.